sync/task_pool_test.go

```diff
@@ -0,0 +1,252 @@
+package gxsync
+
+import (
+	"math/rand"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func newCountTask() (func(), *int64) {
+	var cnt int64
+	return func() {
+		atomic.AddInt64(&cnt, 1)
+	}, &cnt
+}
+
+func TestTaskPool(t *testing.T) {
+	numCPU := runtime.NumCPU()
+	taskCnt := int64(numCPU * numCPU * 10)
+
+	tp := NewTaskPool(
+		WithTaskPoolTaskPoolSize(numCPU/2),
+		WithTaskPoolTaskQueueNumber(numCPU),
+		WithTaskPoolTaskQueueLength(1),
+	)
+
+	task, cnt := newCountTask()
+
+	var wg sync.WaitGroup
+	for i := 0; i < numCPU*numCPU; i++ {
+		wg.Add(1)
+		go func() {
+			for j := 0; j < 10; j++ {
+				tp.AddTask(task)
+			}
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+	tp.Close()
+
+	if taskCnt != *cnt {
+		t.Error("want ", taskCnt, " got ", *cnt)
```

Comment:
@AlexStocks this assert failed sometimes

![image](https://user-images.githubusercontent.com/4499674/74740256-7c770100-5295-11ea-9270-5a9eb4088d41.png)

Reply:
maybe the pool receive done signal before than send channel in 

![image](https://user-images.githubusercontent.com/4499674/74740393-c06a0600-5295-11ea-961c-345b1316f870.png)

Reply:
can you help me check if the test code has errors  ?

Reply:
got the reason by the log https://github.com/dubbogo/gost/pull/25/files#diff-e3fb3a64c85404e71d3fd89a55036c36R144